Core/src/org/sleuthkit/autopsy/directorytree/DirectoryTreeFilterChildren.java

```diff
@@ -204,11 +205,11 @@ private Boolean visitDeep(AbstractAbstractFileNode<? extends AbstractFile> node)
                 if ((childContent instanceof AbstractFile) && ((AbstractFile) childContent).isDir()) {
                     return false;
                 } else {
-                    try {
-                        if (childContent.hasChildren()) {
+                    try{
+                        if(0 < Case.getCurrentCase().getSleuthkitCase().getVisibleContentChildrenCount(childContent)){
```

Comment:
@APriestman and @esaunders , I was actually looking for this query to be made via a companion method to AbstractContentNode.getContentChildren, rather than adding a new, specialized API to SleuthkitCase.  This is for the UI, not for the data model, e.g., the method header in SleuthkitCase refers to lazy (node) loading, which has no meaning in the SleuthKit layer.  So I think we should call SleuthkitCase.executeQuery from a new AbstractContentNode.getVisibleContentChildrenCount method instead.